playground/backend/cmd/server/controller.go

```diff
@@ -529,3 +529,31 @@ func (controller *playgroundController) GetSnippet(ctx context.Context, info *pb
 	}
 	return &response, nil
 }
+
+// GetMetadata returns runner metadata
+func (controller *playgroundController) GetMetadata(_ context.Context, _ *pb.GetMetadataRequest) (*pb.GetMetadataResponse, error) {
+	commitTimestampInteger := func() *int64 {
+		timestamp, err := strconv.ParseInt(BuildCommitTimestamp, 10, 64)
+		if err != nil {
+			logger.Errorf("GetMetadata(): failed to parse BuildCommitTimestamp (\"%s\"): %s", BuildCommitTimestamp, err.Error())
+			return nil
+		}
+		return &timestamp
+	}()
+
+	buildCommitHash := func() *string {
+		if BuildCommitHash == "" {
+			return nil
+		}
+		return &BuildCommitHash
+	}()
```

Comment:
This unreadable code block is due to the use of the optional keyword [here](https://github.com/akvelon/beam/blob/get-metadata-endpoint-playground/playground/api/v1/api.proto#L323) [which is discouraged](https://cloud.google.com/apis/design/proto3). The optional keyword leads to a string* instead of just a string.